### ISSUE TITLE
Usage of typeMappingSource in SqlServerPropertyExtensions.GetValueGenerationStrategy

### ISSUE THREAD

**Issue by dmitry-lipetsk** [dotnet/efcore#26674]
EF6.0

At current time I adding in my provider the support of ValueGenerationStrategy and trying to understand the implementation in exists providers :)

**SqlServerPropertyExtensions code**

https://github.com/dotnet/efcore/blob/b0eaace609c1cbc46e4c32ee732d67195fffb88a/src/EFCore.SqlServer/Extensions/SqlServerPropertyExtensions.cs#L410-L428

As I understood, parameter typeMappingSource contains the hint for work.

This parameter was added in https://github.com/dotnet/efcore/pull/23929

**Why** typeMappingSource does not passed in recursive call (line 424 )?

``` c#
return sharedTableRootProperty.GetValueGenerationStrategy(storeObject) 
```

I think, my question is addressed to @AndriySvyryd.

Thanks for answers.

**Comment by AndriySvyryd** [comment]
That was an oversight. The parameter should be passed in the recursive call as well.

**Comment by AndriySvyryd** [comment]
Fixed in f4aa23208a3891eb16f23db8892218a9efff3dca
